packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/builders/CategoryAwareNotificationBuilder.java

```diff
@@ -0,0 +1,73 @@
+package expo.modules.notifications.notifications.presentation.builders;
+
+import androidx.core.app.RemoteInput;
+import androidx.core.app.NotificationCompat;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.util.Log;
+
+import java.util.List;
+import java.io.IOException;
+import java.util.Collections;
+
+import expo.modules.notifications.notifications.model.NotificationAction;
+import expo.modules.notifications.notifications.model.NotificationCategory;
+import expo.modules.notifications.notifications.model.NotificationContent;
+import expo.modules.notifications.notifications.model.TextInputNotificationAction;
+import expo.modules.notifications.notifications.service.TextInputNotificationResponseReceiver;
+import expo.modules.notifications.notifications.service.SharedPreferencesNotificationCategoriesStore;
+import static expo.modules.notifications.notifications.service.NotificationResponseReceiver.getActionIntent;
+import static expo.modules.notifications.notifications.service.TextInputNotificationResponseReceiver.getActionIntent;
+
+public class CategoryAwareNotificationBuilder extends ExpoNotificationBuilder {
```

Comment:
I am not sure that I get the design fully, but don't we miss setting the badge here, bu extending `ExpoNotificationBuilder` and not `BadgeSettingNotificationBuilder`

Reply:
I extended `ExpoNotificationBuilder` since that was used in `ExpoNotificationsService.java`. Looking at the `BadgeSettingNotificationBuilder` class, I'm not sure where this class is used. 

This is out of scope of this PR, but whether or not using categories, I'm not able to customize the badge count beyond it iterating automatically with each notification, [this method](https://github.com/leolin310148/ShortcutBadger/blob/6d4ce808b0a1abe0ac31dc93d46a78ebecdebfbe/ShortcutBadger/src/main/java/me/leolin/shortcutbadger/ShortcutBadger.java#L103) is throwing (tried default pixel launcher and microsoft launcher)

Reply:
Okay, that is okay for me as of now, but we should double check with @sjchmiela once he is back.